src/OrleansRuntime/Startup/OrleansInternalServices.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Orleans.Runtime.MembershipService;
+using Orleans.Runtime.ReminderService;
+
+namespace Orleans.Runtime.Startup
+{
+    // <summary>
+    // extension methods to register services required internal Orleans methods
+    // </summary>
+    public static class OrleansInternalServices
+    {
+
+        // extension methods to register internal orleans required services
+        public static void RegisterSystemTypes(IServiceCollection serviceCollection)
+        {
+            // Register the system classes and grains in this method.
+            // Note: this method will probably have to be moved out into the Silo class to include internal runtime types.
+            IGrainFactory grainFactory = new GrainFactory();
+            serviceCollection.AddSingleton((GrainFactory)grainFactory);
```

Comment:
Yes totally agree with you here. My first pass is super far away from finished. Just an demonstration of where I want to go.